packages/jest-circus/src/eventHandler.js

```diff
@@ -30,54 +36,70 @@ const handler: EventHandler = (event, state): void => {
     }
     case 'finish_describe_definition': {
       const {currentDescribeBlock} = state;
-      if (!currentDescribeBlock) {
-        throw new Error(
-          `"currentDescribeBlock" has to be there since we're finishing its definition.`,
-        );
-      }
+      invariant(currentDescribeBlock, `currentDescribeBlock mest to be there`);
       if (currentDescribeBlock.parent) {
         state.currentDescribeBlock = currentDescribeBlock.parent;
       }
       break;
     }
     case 'add_hook': {
       const {currentDescribeBlock} = state;
-      const {fn, hookType: type} = event;
-      currentDescribeBlock.hooks.push({fn, type});
+      const {fn, hookType: type, timeout} = event;
+      const parent = currentDescribeBlock;
+      currentDescribeBlock.hooks.push({fn, parent, timeout, type});
       break;
     }
     case 'add_test': {
       const {currentDescribeBlock} = state;
-      const {fn, mode, testName: name} = event;
-      const test = makeTest(fn, mode, name, currentDescribeBlock);
+      const {fn, mode, testName: name, timeout} = event;
+      const test = makeTest(fn, mode, name, currentDescribeBlock, timeout);
       test.mode === 'only' && (state.hasFocusedTests = true);
       currentDescribeBlock.tests.push(test);
       break;
     }
-    case 'test_start': {
-      event.test.startedAt = Date.now();
+    case 'hook_failure': {
+      const {test, describeBlock, error, hook} = event;
+      const {type} = hook;
+
+      if (type === 'beforeAll') {
+        invariant(describeBlock, 'always present for `*All` hooks');
```

Comment:
I think we need to add a prefix to the message? Similar to the contentBlock invariant message